### ISSUE TITLE
Use Legend in Node JS/TS application

### ISSUE THREAD

**Issue by idaromnet** [LegendApp/legend-state#244]
Hello Jay!
I just made an issue instead of keeping the discussion about legend compatibility with Node JS/TS. Still after your update to version 2.1.6 I get the following errors when I use, Legend inside Node ver 21.0.0

ode_modules/@legendapp/state/src/observableInterfaces.d.ts:2:33 - error TS2307: Cannot find module 'react-native-mmkv' or its corresponding type declarations.

2 type MMKVConfiguration = import('react-native-mmkv').MMKVConfiguration;
~~~~~~~~~~~~~~~~~~~

node_modules/@legendapp/state/src/observableInterfaces.d.ts:3:34 - error TS2307: Cannot find module '@react-native-async-storage/async-storage' or its corresponding type declarations.

3 type AsyncStorageStatic = import('@react-native-async-storage/async-storage').AsyncStorageStatic;
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@legendapp/state/src/observableInterfaces.d.ts:4:33 - error TS2307: Cannot find module 'firebase/database' or its corresponding type declarations.

4 type DatabaseReference = import('firebase/database').DatabaseReference;
~~~~~~~~~~~~~~~~~~~

node_modules/@legendapp/state/src/observableInterfaces.d.ts:5:21 - error TS2307: Cannot find module 'firebase/database' or its corresponding type declarations.

5 type Query = import('firebase/database').Query;

**Comment by jmeistrich** [comment]
Could you share a sample repo I can test with? I created a bare node app and tried with versions 18 and 21 and it seems to work ok.

**Comment by idaromnet** [comment]
Hello Jay!
I still get the following errors when my Node JS is version 21 and typescript ver 5.3.3 and legend is "@legendapp/state": "^2.1.6":
I just installed the legend in to my project and just defined a simple state  as the below :
```
const state2 = observable({ test: "not ok" });
console.log(state2.get());
```
then got the following errors:
```

node_modules/@legendapp/state/src/observableInterfaces.d.ts:2:33 - error TS2307: Cannot find module 'react-native-mmkv' or its corresponding type declarations.

2 type MMKVConfiguration = import('react-native-mmkv').MMKVConfiguration;
                                  ~~~~~~~~~~~~~~~~~~~

node_modules/@legendapp/state/src/observableInterfaces.d.ts:3:34 - error TS2307: Cannot find module '@react-native-async-storage/async-storage' or its corresponding type declarations.

3 type AsyncStorageStatic = import('@react-native-async-storage/async-storage').AsyncStorageStatic;   
                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@legendapp/state/src/observableInterfaces.d.ts:4:33 - error TS2307: Cannot find module 'firebase/database' or its corresponding type declarations.

4 type DatabaseReference = import('firebase/database').DatabaseReference;
                                  ~~~~~~~~~~~~~~~~~~~

node_modules/@legendapp/state/src/observableInterfaces.d.ts:5:21 - error TS2307: Cannot find module 'firebase/database' or its corresponding type declarations.

5 type Query = import('firebase/database').Query;
                      ~~~~~~~~~~~~~~~~~~~


Found 4 errors in the same file, starting at: node_modules/@legendapp/state/src/observableInterfaces.d.ts:2
```

**Comment by jmeistrich** [comment]
Could you share a sample repo I can test with? I can't reproduce any errors when testing myself so I don't have a starting point for fixing it.

**Comment by idaromnet** [comment]
https://github.com/idaromnet/legendNode

**Comment by jmeistrich** [comment]
Thanks! Think this should be fixed in 2.1.7.

**Comment by jmeistrich** [comment]
I'll close the issue but please let me know if it's still not right.
